### PR TITLE
feat: Rewrite `automations/methodsMixin` to a composable

### PR DESCRIPTION
**Pull request by iamsivin**
# Pull Request Template

## Description

The PR will replace the usage of `automations/methodsMixin` with the help of `useAutomation()` composable and `automationHelper`

**Files updated**
1. dashboard/routes/dashboard/settings/automation/AddAutomationRule.vue
2. dashboard/routes/dashboard/settings/automation/EditAutomationRule.vue

Fixes https://linear.app/chatwoot/issue/CW-3437/rewrite-automationsmethodsmixin-mixin-to-a-composable

## Type of change

- [x] New feature (non-breaking change which adds functionality)

## How Has This Been Tested?

**Test cases**

1. Test the entire flow of automation.
2. Check adding automation works properly or not with all conditions and actions
3. Check editing automation works properly or not with all conditions and actions

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my code
- [x] I have commented on my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: app/javascript/dashboard/composables/useAutomation.js [hunk 1]
```diff
@@ -0,0 +1,344 @@
...
+  const contacts = useMapGetter('contacts/getContacts');
+  const inboxes = useMapGetter('inboxes/getInboxes');
+  const labels = useMapGetter('labels/getLabels');
+  const teams = useMapGetter('teams/getTeams');
+  const slaPolicies = useMapGetter('sla/getSLA');
+
+  const booleanFilterOptions = computed(() => [
+    { id: true, name: t('FILTER.ATTRIBUTE_LABELS.TRUE') },
+    { id: false, name: t('FILTER.ATTRIBUTE_LABELS.FALSE') },
+  ]);
```
Comment (scmmishra) [inline app/javascript/dashboard/composables/useAutomation.js:39]: Does this need to be a computed? Also why is `id` a boolean, if we're using `id` as a value, it is kind of an anti-pattern, make it a string and have a `value` attribute.

In case the change extends much beyond the scope of this refactor, let's add a TODO and do it in a separate PR

File: app/javascript/dashboard/composables/useAutomation.js [hunk 2]
```diff
@@ -0,0 +1,344 @@
...
+   * Resets an action in the automation.
+   * @param {Object} automation - The automation object to update.
+   * @param {number} index - The index of the action to reset.
+   */
+  const resetAction = (automation, index) => {
+    automation.actions[index].action_params = [];
+  };
+
+  /**
+   * Manifests conditions for the automation.
```
Comment (scmmishra) [inline app/javascript/dashboard/composables/useAutomation.js:155]: This comment could be better, you could mention what does this function actually do and why it is needed

File: app/javascript/dashboard/composables/useAutomation.js
Comment (scmmishra) [inline app/javascript/dashboard/composables/useAutomation.js:1]: Generally for this file, I think we can improve the JSdoc comments, especially for the format and manifest methods

**Review by scmmishra — CHANGES_REQUESTED** [review]

File: app/javascript/dashboard/composables/useAutomation.js (hunk 1 above)
Comment (iamsivin) [reply app/javascript/dashboard/composables/useAutomation.js:39]: @scmmishra We need to have it in the computed property; otherwise, it doesn't render in the multiselect dropdown. The booleanFilterOptions is added to render the attribute type checkbox, and it should include Boolean values. The multiselect tracks values by Id, so if we change it here, it won't work. I think we need to change the others as well.

File: app/javascript/dashboard/composables/useAutomation.js (hunk 2 above)
Comment (iamsivin) [reply app/javascript/dashboard/composables/useAutomation.js]: @scmmishra Done

File: app/javascript/dashboard/composables/useAutomation.js
Comment (iamsivin) [reply app/javascript/dashboard/composables/useAutomation.js:1]: @scmmishra Updated

**Review by scmmishra — APPROVED** [review]

**Comment by scmmishra** [comment]
@vishnu-narayanan need to merge this, the codeclimate issues can be ignored
